pkg/config/patch.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"encoding/json"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/authzed/spicedb-operator/pkg/apis/authzed/v1alpha1"
+)
+
+// TODO: should target take a wildcard? i.e. add labels to all objects?
+// TODO: should patch be an "embedded" yaml? or is it best as a string?
+
+func ApplyPatches[K any](kind string, object K, patches []v1alpha1.Patch) error {
+	for _, p := range patches {
+		if p.Target.Kind == kind {
+			// marshal object to json
+			encoded, err := json.Marshal(object)
+			if err != nil {
+				return err
+			}
+
+			// determine if the patch is a strategic merge or a json6902 patch
+			decoder := utilyaml.NewYAMLOrJSONDecoder(strings.NewReader(p.Patch), 100)
```

Comment:
what is that 100 number?

Reply:
the number of bytes to look ahead to try and determine whether it's json or yaml